Source/WebCore/platform/graphics/PlatformDisplay.cpp

```diff
@@ -457,31 +457,52 @@ const String& PlatformDisplay::drmDeviceFile()
     return m_drmDeviceFile.value();
 }
 
-static String drmRenderNodeFromPrimaryDeviceFile(const String& primaryDeviceFile)
+static void drmForeachDevice(Function<bool(drmDevice*)>&& functor)
 {
-    if (primaryDeviceFile.isEmpty())
-        return { };
-
     drmDevicePtr devices[64];
     memset(devices, 0, sizeof(devices));
 
     int numDevices = drmGetDevices2(0, devices, std::size(devices));
     if (numDevices <= 0)
-        return { };
+        return;
 
-    String renderNodeDeviceFile;
     for (int i = 0; i < numDevices; ++i) {
-        drmDevice* device = devices[i];
+        if (!functor(devices[i]))
+            break;
+    }
+    drmFreeDevices(devices, numDevices);
+}
+
+static String drmFirstRenderNode()
+{
+    String renderNodeDeviceFile;
+    drmForeachDevice([&](drmDevice* device) {
+        if (!(device->available_nodes & (1 << DRM_NODE_RENDER)))
```

Comment:
Shall we add in this condition the DRM_NODE_PRIMARY to the accepted options?

Reply:
No, we only want the render node, not the primary.